### DISCUSSION TITLE
Trigger createDefaultAdmin before runMigrations

### DISCUSSION THREAD

**Discussion by oachoor** [directus/directus#9986] [Feature Requests]
On install, It's not possible to apply a migration on admin-user's entry (`directus_users`), since it concludes a futile attempt to update a non-existing row, and therefore not able to re-run the migration again as it has been saved in `directus_migrations` table.

PullRequest: https://github.com/directus/directus/pull/9895

### To Reproduce

Create a custom-migration:
```
module.exports = {
    async up(knex) {
	await knex("directus_users").update({
		token: "VjocVHdFiz5vG",
		first_name: "Tenant",
		last_name: "One"
	}).where('email', 'tenant1@directus.de')
    },
    async down(knex) {},
};
```
run `directus bootstrap` with ADMIN_EMAIL/ADMIN_PASSWORD env variables.

### Errors Shown

No error, but here's the output:

<img width="527" alt="Screenshot 2021-11-19 at 17 34 13" src="https://user-images.githubusercontent.com/7025827/142658963-fcf2de32-a2f0-4037-a3e0-20f7d4fe1ee7.png">

**Comment by fczuardi** [comment]
I am trying to attempt the same thing, I want to use the docker image, with a default user (admin) and make this first use have a default `token`. When I tried to do that with a custom migration, the update was not possible because this user is only created after all migrations on the bootstrap script.

**Reply by fczuardi** [reply]
oh, nevermind, I just saw the comments on your pull request, I will go to the [same solution you used here](https://github.com/directus/directus/pull/9895#issuecomment-981772147)

thanks!

**Reply by fczuardi** [reply]
well, running all steps didnt quite worked the `roles create` command asked for a Name...

What I ended up doing was hackish but easier, I ran the bootstrap, then `migrate:down`, followed by `migrate:up` :

on docker-compose.yml
```
    command: >
      /bin/sh -c "npx directus bootstrap
      && npx directus database migrate:down
      && npx directus database migrate:up
      && npx directus start"
```

**Reply by oachoor** [reply]
@fczuardi you're right, I've adjusted the code section, but still I think https://github.com/directus/directus/pull/10003 is required to make it work as @rijkvanzanten suggested there.

**Reply by oachoor** [reply]
@fczuardi just tried yours, and that didn't work either when your migration is not the latest, it looks like migration down & up are only affecting the latest one, please make sure your custom-migration's date is the most recent.

<img width="541" alt="Screenshot 2021-12-03 at 16 13 42" src="https://user-images.githubusercontent.com/7025827/144626400-15830a40-d775-4355-b85c-ea03d27a1b9e.png">

**Reply by rijkvanzanten** [reply]
#10003 was merged and will be released today. That should allow you to do all the bootstrap steps manually, therefore allowing you to control the order of when the schema snapshot is applied

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
